src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -97,10 +97,14 @@ def failedJobs(self, failedJobs, useMask=True):
             for fInfo in inputFiles:
                 if fInfo["merged"]:
```

Comment:
Haaa... it was this merged value that is a string, either '0' or '1', so it evaluates always to True :(